crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -546,8 +542,50 @@ async fn execute_transactions(
 
     authority_state.transaction_manager.enqueue(txns).await?;
 
-    authority_state
+    let effects = authority_state
         .database
         .notify_read_effects(tx_digests)
-        .await
+        .await?;
+
+    assert_eq!(
+        effects.len(),
+        num_digests,
+        "Missing txn effects from transaction manager"
+    );
+    Ok(effects)
```

Comment:
Maybe just return Ok(()) instead. The returned effects do not seem to be used. And if anyone uses them, they can be in a different order than execution_digests which will be unexpected.